apps/web/src/components/templates/TemplatePageHeader.tsx

```diff
@@ -0,0 +1,105 @@
+import { Center, Container, Grid, Group } from '@mantine/core';
+import { useState } from 'react';
+import { Button, colors, Switch, Title, Text } from '../../design-system';
+import { ArrowLeft } from '../../design-system/icons';
+import { ActivePageEnum } from '../../pages/templates/editor/TemplateEditorPage';
+import { useEnvController } from '../../store/use-env-controller';
+import { When } from '../utils/When';
+import { EditorPreviewSwitch } from './EditorPreviewSwitch';
+import { useStatusChangeControllerHook } from './use-status-change-controller.hook';
+import { useTemplateController } from './use-template-controller.hook';
+
+const Header = ({ activePage, editMode }: { editMode: boolean; activePage: ActivePageEnum }) => {
+  if (activePage === ActivePageEnum.Settings) {
+    return <>{editMode ? 'Edit Template' : 'Create new template'}</>;
+  }
+  if (activePage === ActivePageEnum.Workflow) {
+    return <>{'Workflow Editor'}</>;
+  }
+
+  if (activePage === ActivePageEnum.sms) {
+    return <>{'Edit SMS Template'}</>;
+  }
+
+  if (activePage === ActivePageEnum.email) {
+    return <>{'Edit Email Template'}</>;
+  }
+
+  if (activePage === ActivePageEnum.in_app) {
+    return <>{'Edit Notification Template'}</>;
+  }
+
+  return <>{editMode ? 'Edit Template' : 'Create new template'}</>;
+};
+
+interface Props {
+  templateId: string;
+  loading: boolean;
+  disableSubmit: boolean;
+  setActivePage: any;
```

Comment:
```suggestion
  setActivePage:(activePage: ActivePageEnum) => void;
```

Can it be like this maybe to avoid any usage?